homeassistant/components/climate/zha.py

```diff
@@ -147,13 +150,17 @@ def async_update(self):
             _LOGGER.warning("No result received")
             return
 
-        self._heating_power = int(result.get('pi_heating_demand'))
+        try:
+            self._heating_power = int(result['pi_heating_demand'])
+        except (TypeError, KeyError):
+            _LOGGER.warning("Can not get heating power value")
+            self._heating_power = None
 
-        raw_mode = result.get('setpoint_mode_manuf_specific', {})
-        if raw_mode not in self._modes:
+        raw_mode = result.get('setpoint_mode_manuf_specific')
+        if raw_mode not in OPERATIONS:
             _LOGGER.warning("Mode %s not found in known modes %s",
-                            raw_mode, self._modes)
-        self._current_operation = self._modes.get(raw_mode)
+                            raw_mode, OPERATIONS)
+        self._current_operation = OPERATIONS[raw_mode]
```

Comment:
You still set the operation, even if it is not a possible operation - is that intended? Have you tested what that does to the front end? I guess no operation mode will be shown in the front end in this case.

Reply:
Better use `dict.get` here like it was before. Then it will return `None` if `raw_mode` is missing in `OPERATIONS`.